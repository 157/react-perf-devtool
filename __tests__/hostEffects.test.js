var MEASURES = require('../samples/measures')

var {
  getCommitHostEffectsTime,
  getTotalEffects
} = require('../src/shared/hostEffects')
var { getTotalTime } = require('../src/shared/totalTime')

describe('Committing Host Effects', () => {
  test('Sanity', () => {
    expect(getCommitHostEffectsTime(MEASURES)).toMatchSnapshot()
  })

  test('Total time taken in committing host effects', () => {
    expect(getTotalTime(getCommitHostEffectsTime(MEASURES))).toMatchSnapshot()
  })

  test('Total effects', () => {
    expect(
      getTotalEffects(getCommitHostEffectsTime(MEASURES))
    ).toMatchSnapshot()
  })
})
