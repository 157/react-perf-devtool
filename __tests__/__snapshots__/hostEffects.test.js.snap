// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Committing Host Effects Sanity 1`] = `
Object {
  "Committing Host Effects": Object {
    "timeSpent": Array [
      1.14,
      0.09,
    ],
    "totalEffects": Array [
      7,
      1,
    ],
  },
}
`;

exports[`Committing Host Effects Total effects 1`] = `8`;

exports[`Committing Host Effects Total time taken in committing host effects 1`] = `1.23`;
